src/collapse/collapse.ts

```diff
@@ -27,8 +32,31 @@ export class NgbCollapse {
 
   @Output() ngbCollapseChange = new EventEmitter<boolean>();
 
+  /**
+   * An event emitted when the collapse element is shown, after the transition. It has no payload.
+   */
+  @Output() shown = new EventEmitter<void>();
+
+  /**
+   * An event emitted when the collapse element is hidden, after the transition. It has no payload.
+   */
+  @Output() hidden = new EventEmitter<void>();
+
+
   constructor(private _element: ElementRef, config: NgbCollapseConfig) { this.animation = config.animation; }
 
+  ngOnInit() {
+    const {classList} = this._element.nativeElement;
+    classList.add('collapse');
```

Comment:
I think this one we should add via the `'class': 'collapse'` binding as before, what's the point doing it imperatively, if we could do this declaratively with Angular...

Reply:
I would prefer not merged what is done by binding and what is done by the transition engine. 

As far as remember, the collapse class was added back just after starting the transition (direction: show), because `collapsed` is true. The result is there is `collapsing`and `collapse`at the same time in the dom.

I will check, but I thing the tests in this PR prevent this, so it should fail.